cpp/src/arrow/ipc/message.cc

```diff
@@ -279,8 +280,37 @@ std::string FormatMessageType(MessageType type) {
   return "unknown";
 }
 
+Status ReadFieldsSubset(int64_t offset, int32_t metadata_length,
+                        io::RandomAccessFile* file,
+                        const FieldsLoaderFunction& fields_loader,
+                        const std::shared_ptr<Buffer>& metadata, int64_t required_size,
+                        std::shared_ptr<Buffer>& body) {
+  const flatbuf::Message* message = nullptr;
+  uint8_t continuation_metadata_size = sizeof(int32_t) + sizeof(int32_t);
+  // skip 8 bytes (32-bit continuation indicator + 32-bit little-endian length prefix)
+  RETURN_NOT_OK(internal::VerifyMessage(metadata->data() + continuation_metadata_size,
+                                        metadata->size() - continuation_metadata_size,
+                                        &message));
+
+  auto batch = message->header_as_RecordBatch();
```

Comment:
Should we return an error if the message isn't a record batch?

Reply:
You are right. Now I added an error checking like we did in `reader.cc`